.github/workflows/ci-build.yml

```diff
@@ -1,27 +1,73 @@
-name: Build and Tests
+name: 'Build and Test' 
 
 on:
   push:
-    branches: [ main ]
   pull_request:
     branches: [ main ]
 
 jobs:
-  build:
+  Build:
+
+    runs-on: ubuntu-latest
+    env: 
+      VSTEST_CONNECTION_TIMEOUT: 900
 
+    strategy:
+      matrix:
+        dotnet-version: [ 'netcoreapp3.1', 'net5', 'net6' ] 
+         
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '3.1.x'
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '5.0.x'
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '6.0.x'
+      - name: Install dependencies
+        run: dotnet restore ./src/
+      - name: Build solution
+        run: dotnet build ./src/ --no-restore --framework ${{matrix.dotnet-version}}
+  
+  Test:
```

Comment:
Should we be testing in a different job? Like this we will be building the solution twice, if it was one jobs then it would only have to build once and take less time to do so shortening the feedback loop. If you have a great reason to use two jobs then it might be an idea to cache the built outputs and then run the tests using those.

Reply:
Simple answer is that I was probably too lazy.
I will fix it.